test/version_test.exs

```diff
@@ -33,86 +35,154 @@ defmodule GitOps.Test.VersionTest do
+  end
+
+  test "a release candidate starts at 0 if requested on patch" do
+    assert new_version("0.1.0", [patch()], rc: true) == "0.1.1-rc0"
+  end
+
+  test "a release candidate starts at 0 if requested on minor" do
+    assert new_version("0.1.0", [minor()], rc: true) == "0.2.0-rc0"
+  end
+
+  test "a release candidate starts at 0 if requested on break" do
+    assert new_version("0.1.0", [break()], rc: true) == "1.0.0-rc0"
   end
 
-  test "a release candidate starts at 0 if requested" do
-    assert new_version(["0.1.0"], [patch()], rc: true) == "0.1.1-rc0"
+  test "a release candidate increments by one on patch" do
+    assert new_version("0.1.1-rc0", [patch()], rc: true) == "0.1.1-rc1"
   end
 
-  test "a release candidate increments by one as long as the version would normally change" do
-    assert new_version(["0.1.0", "0.1.1-rc0"], [patch()], rc: true) == "0.1.1-rc1"
+  test "a release candidate resets on minor" do
+    assert new_version("0.1.1-rc0", [minor()], rc: true) == "0.2.0-rc0"
   end
 
-  test "a release candidate resets if the version would change more than it originally did" do
-    assert new_version(["0.1.0", "0.1.1-rc0"], [break()], rc: true) == "1.0.0-rc0"
+  test "a release candidate resets on major" do
+    assert new_version("0.1.1-rc0", [break()], rc: true) == "1.0.0-rc0"
   end
 
   test "a release candidate raises correctly when it would not change" do
     assert_raise RuntimeError, ~r/No changes should result in a new release version./, fn ->
-      new_version(["0.1.0", "0.1.1-rc0"], [chore()], rc: true)
+      new_version("0.1.1-rc0", [chore()], rc: true)
+    end
+  end
+
+  test "invalid rc number version raises" do
+    assert_raise RuntimeError, ~r/Found an rc version that could not be parsed/, fn ->
+      new_version("0.1.1-rca", [patch()], rc: true)
+    end
+  end
+
+  test "invalid rc label version raises" do
+    assert_raise RuntimeError, ~r/Found an rc version that could not be parsed/, fn ->
+      new_version("0.1.1-releasecandidate", [patch()], rc: true)
+    end
+  end
+
+  test "no prefix on version raises" do
+    assert_raise ArgumentError, ~r/Expected: .+ to be parseable as a version, but it was not/, fn ->
+      new_version("v0.1.1", [patch()], rc: true, prefix: "vv")
     end
   end
 
-  test "if a prefix is configured, it is ignored when searching for a tag" do
-    assert new_version(["v0.1.1", "2.0.0"], [break()], prefix: "v") == "v1.0.0"
+  test "last valid non rc is found correctly without prefixes" do
+    versions = ["0.0.1", "0.1.0-rc0", "0.1.0", "0.2.0-alpha"]
+    last_rc = Version.last_valid_non_rc_version(versions, "")
+
+    assert last_rc == "0.1.0"
+  end
+
+  test "last valid non rc is found correctly with prefixes" do
+    versions = ["v0.0.1", "v0.1.0-rc0", "0.1.0", "0.2.0-alpha"]
+    last_rc = Version.last_valid_non_rc_version(versions, "v")
+
+    assert last_rc == "v0.0.1"
   end
+
+  # QUESTION: is this supposed to be the functionality?
```

Comment:
Question

Reply:
I don't remember my initial intent with this, to be honest. I say for now we just rename the function to what it actually is, which is just `last_version_later_than`. Afterwards we should just revisit the logic for how we handle the `rc` flag. I think we'll find its much simpler than I made it :) (keep in mind I hacked this together in like two afternoons :/ )

Reply:
Actually its: `latest_version_greater_than`

Reply:
Sounds good! Yeah, I figured there were definitely some simplifications to be made